fail-build-on-error not respected if annotation is too large

The `fail-build-on-error` flag is ignored if the annotation is too large. It would still be beneficial in our case to fail the build even if there is no annotation instead of wrongfully passing the build. Here is a small patch that would solve this issue.

```
diff --git a/hooks/command b/hooks/command
index 2e03f21..6bfbe65 100755
--- a/hooks/command
+++ b/hooks/command
@@ -48,13 +48,12 @@ if grep -q "<details>" "$annotation_path"; then
   if ! check_size; then
       echo "--- :warning: Failures too large to annotate"
       echo "The failures are too large to create a build annotation. Please inspect the failed JUnit artifacts manually."
-      exit 0
+  else
+    echo "--- :buildkite: Creating annotation"
+    # shellcheck disable=SC2002
+    cat "$annotation_path" | buildkite-agent annotate --context junit --style error
   fi

-  echo "--- :buildkite: Creating annotation"
-  # shellcheck disable=SC2002
-  cat "$annotation_path" | buildkite-agent annotate --context junit --style error
-
   if [[ "${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAIL_BUILD_ON_ERROR:-false}" =~ (true|on|1) ]]
   then
     echo "--- :boom: Failing build due to error"
```

--- a/hooks/command
+++ b/hooks/command
@@ -46,14 +46,13 @@ cat "$annotation_path"
 if grep -q "<details>" "$annotation_path"; then
 
   if ! check_size; then
-      echo "--- :warning: Failures too large to annotate"
-      echo "The failures are too large to create a build annotation. Please inspect the failed JUnit artifacts manually."
-      exit 0
+    echo "--- :warning: Failures too large to annotate"
+    echo "The failures are too large to create a build annotation. Please inspect the failed JUnit artifacts manually."
+  else
+    echo "--- :buildkite: Creating annotation"
+    # shellcheck disable=SC2002
+    cat "$annotation_path" | buildkite-agent annotate --context "${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_CONTEXT:-junit}" --style error
   fi
-
-  echo "--- :buildkite: Creating annotation"
-  # shellcheck disable=SC2002
-  cat "$annotation_path" | buildkite-agent annotate --context "${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_CONTEXT:-junit}" --style error
 
   if [[ "${BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAIL_BUILD_ON_ERROR:-false}" =~ (true|on|1) ]]
   then

--- a/tests/command.bats
+++ b/tests/command.bats
@@ -149,3 +149,32 @@ load "$BATS_PATH/load.bash"
   unstub du
   unstub mktemp
 }
+
+@test "returns an error if fail-build-on-error is true and annotation is too large" {
+  export BUILDKITE_PLUGIN_JUNIT_ANNOTATE_ARTIFACTS="junits/*.xml"
+  export BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAIL_BUILD_ON_ERROR=true
+
+  artifacts_tmp="tests/tmp/$PWD/junit-artifacts"
+  annotation_tmp="tests/tmp/$PWD/junit-annotation"
+
+  stub mktemp \
+    "-d junit-annotate-plugin-artifacts-tmp.XXXXXXXXXX : mkdir -p $artifacts_tmp; echo $artifacts_tmp" \
+    "-d junit-annotate-plugin-annotation-tmp.XXXXXXXXXX : mkdir -p $annotation_tmp; echo $annotation_tmp"
+
+  # 1KB over the 1MB size limit of annotations
+  stub du "-k /plugin/tests/tmp//plugin/junit-annotation/annotation.md : echo 1025 /plugin/tests/tmp//plugin/junit-annotation/annotation.md"
+  
+  stub buildkite-agent "artifact download junits/*.xml /plugin/tests/tmp//plugin/junit-artifacts : echo Downloaded artifacts"
+
+  stub docker "--log-level error run --rm --volume /plugin/tests/tmp//plugin/junit-artifacts:/junits --volume /plugin/hooks/../ruby:/src --env BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN= --env BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_FORMAT= ruby:2.7-alpine ruby /src/bin/annotate /junits : echo '<details>Failure</details>'"
+
+  run "$PWD/hooks/command"
+
+  assert_failure
+
+  assert_output --partial "Failures too large to annotate"
+
+  unstub mktemp
+  unstub buildkite-agent
+  unstub docker
+}

